### PR TITLE
Add `GenericTargetFactory` and related configuration

### PR DESCRIPTION
**Pull request by jdreesen**
Adds a `GenericTargetFactory`, which means you can skip writing your own `TargetTypeFactory` and just configure your `target` type if nothing more is needed than `new MyTargetType()`.

### CODE REVIEW

**Comment by jdreesen** [comment]
Converted to draft until #37 is merged.

File: src/DependencyInjection/NeustaConverterExtension.php
Comment (mike4git) [inline src/DependencyInjection/NeustaConverterExtension.php]: I'm already trying to separate that into different strategy handlers - otherwise this configuration handling is not readable anymore.
Especially this ternary-operator-not-null-game is here very confusing.

Comment (jdreesen) [reply src/DependencyInjection/NeustaConverterExtension.php]: Please don't put too much energy into it! These files tend to look very ugly and usually do not conform to normal coding standards. Take a look at the Symfony files, for example: https://github.com/symfony/symfony/blob/6.3/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

I've also never seen them split into separate files, so please let's stick to how Symfony and the other bundles do it.

Comment (jdreesen) [reply src/DependencyInjection/NeustaConverterExtension.php]: @mike4git what about writing it like this? Would that be an improvement for you?

```suggestion
        $targetFactoryId = $config['target_factory'] ?? "{$id}.target_factory";

        if (!isset($config['target_factory'])) {
            $container->register($targetFactoryId, GenericTargetFactory::class)
                ->setArguments([
                    '$type' => new Reference($config['target']),
                ]);
        }

```

**Comment by mike4git** [comment]
I think my idea about "[Single-level-of-abstraction](https://clean-code-developer.de/die-grade/orangener-grad/#Single_Level_of_Abstraction_SLA)" force to implement several private methods which would be readable and "documents" what would be done in each step:
```
$this->registerTargetFactory(...);
$this->registerConverter(...);
...
```
and of course there could be some more private "helper" methods which checks if a field is set and defined or do the registration.

**Comment by jdreesen** [comment]
This is currently on hold until #41 is done.

**Review by mike4git — APPROVED** [review]
Thx

**Pull request merged**
